### DISCUSSION TITLE
TMax in FAST Library

### DISCUSSION THREAD

**Discussion by rafmudaf** [OpenFAST/openfast#1193] [General]
The FAST_Sizes routine in the FAST Library was recently changed so that the calling code could use the TMax specified in the input file (see https://github.com/OpenFAST/openfast/pull/962). Previously, the value of TMax given in FAST_Sizes would be used over the value in the input file. In an attempt to preserve both functionalities, #962 added optional arguments to FAST_Sizes. Since this is a C++ feature and not available in C, compiling for Simulink does not work. A workaround is added in https://github.com/OpenFAST/openfast/pull/1180 to use the optional arguments when compiling for C++ and maintain the previous API when compiling for C. This is not a sustainable solution since now the API is different depending on the language.

Here are a few key steps in going from `FAST_Sizes` to `FAST_init`:

- https://github.com/OpenFAST/openfast/blob/main/modules/openfast-library/src/FAST_Library.f90#L117
- https://github.com/OpenFAST/openfast/blob/main/modules/openfast-library/src/FAST_Library.f90#L145
- https://github.com/OpenFAST/openfast/blob/main/modules/openfast-library/src/FAST_Subs.f90#L50
- https://github.com/OpenFAST/openfast/blob/main/modules/openfast-library/src/FAST_Subs.f90#L204

I don't have a great idea for handling both behaviors. It seems the feature to use TMax from the subroutine is tied directly to the Simulink functionality, while using TMax from the input file is required for the Python integration to WEIS.

**Comment by bjonkman** [comment]
A couple ideas I have: 
1. add a logical flag to the API to determine if the user wants to update TMax or not
2. initialize TMax in the Python integration to a negative number if you don't want to override the input file value. 

Because Simulink has its own TMax, we decided to override the TMax from the OpenFAST input file, but that could also be modified.
